stripe-core/src/main/java/com/stripe/android/core/networking/DefaultStripeNetworkClient.kt

```diff
@@ -16,46 +16,54 @@ class DefaultStripeNetworkClient @JvmOverloads constructor(
     private val workContext: CoroutineContext = Dispatchers.IO,
     private val connectionFactory: ConnectionFactory = ConnectionFactory.Default,
     private val retryDelaySupplier: RetryDelaySupplier = RetryDelaySupplier(),
+    private val retryInterceptor: RetryInterceptor = RetryInterceptor(),
```

Comment:
Seems like there's no interest so far on providing custom per-request logic besides http code checks, so setting a default global retry interceptor (that just checks for common connection errors + existing retriable http codes in the request)

Potentially a request could define its own retry logic in the future if needed.